File: DSA_Questions/src/com/dsa_questions/Balanced_Brackets.java
package com.dsa_questions;

import java.util.Stack;

public class Balanced_Brackets {

public static String Check(String str){
        
        char[] ch = str.toCharArray();
        
        Stack<Character> stack = new Stack<>();
        
        if(ch[0] == ')' || ch[0] == '}' || ch[0] == ']'){
                
                return "not balanced";
                
            }
        
        for(int i = 0;i < ch.length;i++){

            
            if(ch[i] == '(' || ch[i] == '{' || ch[i] == '['){
                
                stack.push(ch[i]);
                
            }else{
                
                if(stack.peek() == '(' && ch[i] == ')'){
                    
                    stack.pop();
                    
                    
                }else if(stack.peek() == '{' && ch[i] == '}'){
                    
                    stack.pop();
                    
                }else if(stack.peek() == '[' && ch[i] == ']'){
                    
                    stack.pop();
                    
                }
                else{
                    
                    return "not balanced";
                    
                }
                
            }
            
        }

        if(stack.isEmpty()){
            
            return "balanced";
            
        }else{
            
            return "not balanced";
            
        }
        
    }
	
}

File: DSA_Questions_Python/Balanced_Brackets.py
def check(str):
    stack = []

    if not str or str[0] == ')' or str[0] == '}' or str[0] == ']':
        return "not balanced"

    for ch in str:
        if ch == '(' or ch == '{' or ch == '[':
            stack.append(ch)
        else:
            if not stack:
                return "not balanced"
            if (stack[-1] == '(' and ch == ')') or \
               (stack[-1] == '{' and ch == '}') or \
               (stack[-1] == '[' and ch == ']'):
                stack.pop()
            else:
                return "not balanced"

    if not stack:
        return "balanced"
    else:
        return "not balanced"